Assets/Resources/MagicItemTemplates.txt

```diff
@@ -1,1359 +1,3070 @@
-            26,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1
+            {
+                "type": "SpecialArtifactEffect",
+                "param": 0
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            }
         ],
         "uses": 1500,
-        "unknown1": 50000,
-        "material": 0,
-        "unknown2": 0
+        "value": 50000,
+        "material": 0
     },
     {
-        "position": 66,
+        "index": 66,
         "name": "The Mehrunes Razor",
         "type": "ArtifactClass1",
         "group": 3,
         "groupIndex": 0,
         "enchantments": [
-            282,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1,
-            -1
+            {
+                "type": "SpecialArtifactEffect",
+                "param": 1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            },
+            {
+                "type": "None",
+                "param": -1
+            }
         ],
         "uses": 1500,
-        "unknown1": 50000,
-        "material": 0,
-        "unknown2": 2048
+        "value": 50000,
+        "material": 8
     },
     {
-        "position": 128,
+        "index": 128,
         "name": "The Mace of Molag Bal",
         "type": "ArtifactClass1",
         "group": 3,
         "groupIndex": 11,
         "enchantments": [
-            538,
```

Comment:
I've noticed the first effect param is condensed to just "SpecialArtifactEffect" and discards unique value read from the binary. This data is exported directly to JSON from MAGIC.DEF, and I had intended to hold exported data in its raw form.

This value will also help identify which "SpecialArtifactEffect" should be deployed. With these changes, that identifying value read from MAGIC.DEF is lost and the artifact items no longer have a unique effect value for that artifact. I feel this will make it more difficult to map these values back the implemented effect I had intended.

I would prefer the raw "enchantments" data is preserved, even if in just another form such as "legacyEnchantments". What are your thoughts?

Reply:
No data is being discarded! The enchantment for the Mace of Molag Bal is 1A 02, which was being read as it's decimal form, 538. But classic treats it as 2 values, the first (1A) is the wider category of enchantment (which I called "type") and the second (02) is interpreted depending on the enchantment category. In this case 1A (26 in decimal) is the category for unique artifact effects like the mace of molag bal absorbing strength,  so I called it "SpecialArtifactEffect". The second value (which I called "param") defines which of the unique artifact effects it is. "0" is the Masque of Clavicus (Vile), "1" is The Mehrunes Razor, "2" is the Mace of Molag Bal effect, etc.)

We could get rid of the enum if you'd prefer it to stay as numbers, (e.g. "26, 0") but I don't think that what is considered by classic like: Artifact Effect 0, Artifact Effect 1, Artifact Effect 2 should be stored as "26", "282", "538".

Reply:
No worries, thanks for explaining. :)